<a href="https://colab.research.google.com/github/shivanalyst/ML-Projects/blob/main/ML_Assignment_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Machine Learning With Python: Linear Regression With One Variable**

**Problem Statement:** Predict canada's per capita income in year 2020. Using this build a regression model and predict the per capita income of canadian citizens in year 2020

**Steps to be followed for solving the Problem Statement:**

1.   Importing the Libraries
2.   Loading the Dataset
1.   Defining Dependent and Independent Variables
2.   Perform Exploratory data analysis
1.   Build a Linear Regression Model
2.   Predict the per capita income for 2020.







In [ ]:
import pandas as pd
import numpy as np
from sklearn import linear_model
import matplotlib.pyplot as plt

In [36]:
df = pd.read_csv('canada_per_capita_income.csv')
df

year        income
0   1970   3399.299037
1   1971   3768.297935
2   1972   4251.175484
3   1973   4804.463248
4   1974   5576.514583
5   1975   5998.144346
6   1976   7062.131392
7   1977   7100.126170
8   1978   7247.967035
9   1979   7602.912681
10  1980   8355.968120
11  1981   9434.390652
12  1982   9619.438377
13  1983  10416.536590
14  1984  10790.328720
15  1985  11018.955850
16  1986  11482.891530
17  1987  12974.806620
18  1988  15080.283450
19  1989  16426.725480
20  1990  16838.673200
21  1991  17266.097690
22  1992  16412.083090
23  1993  15875.586730
24  1994  15755.820270
25  1995  16369.317250
26  1996  16699.826680
27  1997  17310.757750
28  1998  16622.671870
29  1999  17581.024140
30  2000  18987.382410
31  2001  18601.397240
32  2002  19232.175560
33  2003  22739.426280
34  2004  25719.147150
35  2005  29198.055690
36  2006  32738.262900
37  2007  36144.481220
38  2008  37446.486090
39  2009  32755.176820
40  2010  38420.522890
41  2011  42334.711210
42  2012  42665.255970
43  2013  42676.468370
44  2014  41039.893600
45  2015  35175.188980
46  2016  34229.193630

In [37]:
# This command displays whether any NUll value is present or not
df.isnull( )

year  income
0   False   False
1   False   False
2   False   False
3   False   False
4   False   False
5   False   False
6   False   False
7   False   False
8   False   False
9   False   False
10  False   False
11  False   False
12  False   False
13  False   False
14  False   False
15  False   False
16  False   False
17  False   False
18  False   False
19  False   False
20  False   False
21  False   False
22  False   False
23  False   False
24  False   False
25  False   False
26  False   False
27  False   False
28  False   False
29  False   False
30  False   False
31  False   False
32  False   False
33  False   False
34  False   False
35  False   False
36  False   False
37  False   False
38  False   False
39  False   False
40  False   False
41  False   False
42  False   False
43  False   False
44  False   False
45  False   False
46  False   False

In [38]:
# This command helps in understanding the size of the dataset
df.shape

(47, 2)

In [39]:
new_df = df.drop('income',axis='columns')
new_df

year
0   1970
1   1971
2   1972
3   1973
4   1974
5   1975
6   1976
7   1977
8   1978
9   1979
10  1980
11  1981
12  1982
13  1983
14  1984
15  1985
16  1986
17  1987
18  1988
19  1989
20  1990
21  1991
22  1992
23  1993
24  1994
25  1995
26  1996
27  1997
28  1998
29  1999
30  2000
31  2001
32  2002
33  2003
34  2004
35  2005
36  2006
37  2007
38  2008
39  2009
40  2010
41  2011
42  2012
43  2013
44  2014
45  2015
46  2016

In [40]:
Income = df.income
Income

0      3399.299037
1      3768.297935
2      4251.175484
3      4804.463248
4      5576.514583
5      5998.144346
6      7062.131392
7      7100.126170
8      7247.967035
9      7602.912681
10     8355.968120
11     9434.390652
12     9619.438377
13    10416.536590
14    10790.328720
15    11018.955850
16    11482.891530
17    12974.806620
18    15080.283450
19    16426.725480
20    16838.673200
21    17266.097690
22    16412.083090
23    15875.586730
24    15755.820270
25    16369.317250
26    16699.826680
27    17310.757750
28    16622.671870
29    17581.024140
30    18987.382410
31    18601.397240
32    19232.175560
33    22739.426280
34    25719.147150
35    29198.055690
36    32738.262900
37    36144.481220
38    37446.486090
39    32755.176820
40    38420.522890
41    42334.711210
42    42665.255970
43    42676.468370
44    41039.893600
45    35175.188980
46    34229.193630
Name: income, dtype: float64

In [42]:
# Create linear regression object
reg = linear_model.LinearRegression()
reg.fit(new_df,Income)

LinearRegression()

In [43]:
reg.predict([[2020]])

/usr/local/lib/python3.8/dist-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([41288.69409442])

Problem is solved. 
per capita income of Canadians in 2020 : 41288.69409442